utility/session.go

```diff
@@ -1,157 +1,266 @@
-//
-// 2) calls `pseudoRandomSelection(servicers, numberOfNodesPerSession)`
-func (s *session) selectSessionServicers(numServicers int) ([]*coreTypes.Actor, types.Error) {
-	// IMPORTANT: This function is for behaviour illustrative purposes only and implementation may differ.
-	return nil, nil
+// hydrateSessionFishermen finds the fishermen that are staked at the session height and populates the session with them
+func (s *sessionHydrator) hydrateSessionFishermen() error {
+	// number of fishermen per session at this height
+	numFishermen, err := s.readCtx.GetIntParam(types.FishermanPerSessionParamName, s.session.SessionHeight)
+	if err != nil {
+		return err
+	}
+
+	// returns all the staked fishermen at this session height
+	fishermen, err := s.readCtx.GetAllFishermen(s.session.SessionHeight)
+	if err != nil {
+		return err
+	}
+
+	// OPTIMIZE: Consider updating the persistence module so a single SQL query can retrieve all of the actors at once.
+	candidateFishermen := make([]*coreTypes.Actor, 0)
+	for _, fisher := range fishermen {
+		// Sanity check the fisher is not paused, jailed or unstaking
+		if !(fisher.PausedHeight == -1 && fisher.UnstakingHeight == -1) {
+			return fmt.Errorf("hydrateSessionFishermen should not have encountered a paused or unstaking fisherman: %s", fisher.Address)
+		}
+
+		// TODO(#697): Filter by geo-zone
+
+		// OPTIMIZE: If this was a map[string]struct{}, we could have avoided the loop
+		if slices.Contains(fisher.Chains, s.session.RelayChain) {
+			candidateFishermen = append(candidateFishermen, fisher)
+		}
+	}
+
+	s.session.Fishermen = pseudoRandomSelection(candidateFishermen, numFishermen, s.sessionIdBz)
+	return nil
 }
 
-// uses the current 'world state' to determine the fishermen in the session
-// 1) get an ordered list of the public keys of fishermen who are:
-//   - actively staked
-//   - staked within geo-zone  (or closest geo-zones)
-//   - staked for relay-chain
-//
-// 2) calls `pseudoRandomSelection(fishermen, numberOfFishPerSession)`
-func (s *session) selectSessionFishermen(numFishermen int) ([]*coreTypes.Actor, types.Error) {
-	// IMPORTANT: This function is for behaviour illustrative purposes only and implementation may differ.
-	return nil, nil
+// pseudoRandomSelection returns a random subset of the candidates.
+// DECIDE: We are using a `Go` native implementation for a pseudo-random number generator. In order
+// for it to be language agnostic, a general purpose algorithm needs ot be used.
+func pseudoRandomSelection(candidates []*coreTypes.Actor, numTarget int, sessionId []byte) []*coreTypes.Actor {
+	// If there aren't enough candidates, return all of them
+	if numTarget > len(candidates) {
+		logger.Global.Warn().Msgf("pseudoRandomSelection: numTarget (%d) is greater than the number of candidates (%d)", numTarget, len(candidates))
+		return candidates
+	}
+
+	// Take the first 8 bytes of sessionId to use as the seed
+	seed := int64(binary.BigEndian.Uint64(crypto.SHA3Hash(sessionId)[:8]))
```

Comment:
Is there a specific reason why `BitEndian` here and `LittleEndian` when serializing the height? Is this part of a specification?

(I always forget which is conventional in which context. A quick skim of [Endianness](https://en.wikipedia.org/wiki/Endianness) leads me to believe networking is conventionally big-endian, and my recollection is that the dichotomy was  between networking and memory.)

Reply:
> Is there a specific reason why BitEndian here and LittleEndian when serializing the height? Is this part of a specification?

Tbh, I always get mixed up as well. Let's talk about hex at some point.